tests/testthat/test-target_market_share.R

```diff
@@ -170,6 +170,20 @@ test_that("w/ NAs in crucial columns, errors with informative message", {
   expect_error_crucial_NAs_scenario("smsp")
 })
 
+test_that("filters and warns when input-data has NAs", {
+  matched <- fake_matched()
+  ald <- fake_ald(production = c(1, NA))
+  scenario <- fake_scenario()
+
+  expect_warning(
+    target_market_share(
+      matched,
+      ald,
+      scenario,
+      region_isos_stable,
+      class = "na_input_data"))
```

Comment:
Test looks great, my only comment is that, for `target_sda`, the class of the warning is `na_emission_factor`. 
For symmetry, it would be nice if the class here was similar, ie, `na_production`.

Reply:
I had thought of that too but I was unsure of the class because from my understanding, both `production` and `emission_factor` could have `NA`s, right? This is why I chose "na_input_data", to include both. I do however understand the point about symmetry and I'm happy to make the change :)

Reply:
Ya, so we have some options here. 
The easiest would be to write two different warnings, one for `target_market_share` and one for `target_sda`, and give them the class `na_production` and `na_emission_factor`. But this is sorta WET code. 

The other option would be to find the common denominator between the behaviour we want in both function. For example, I think `na_input_data` is too general, since it isn't the behaviour we expect for NAs in ALL input datasets, it is only for two very special columns in the `ald` input dataset.

Reply:
Totally agree! Especially since I can then write a function, e.g. like `check_no_value_is_missing` in `utils.R` but with a warning and call that function in both `target_sda` and `target_market_share`

Reply:
something like `class = "na_input_ald"` or  `class = "na_ald_prod_ef"` ?